nexus/reconfigurator/planning/src/mgs_updates/test_helpers.rs

```diff
@@ -0,0 +1,718 @@
+    ArtifactHash([5; 32]);
+/// Hash of fake artifact for fake sidecar-c SP
+pub(super) const ARTIFACT_HASH_SP_SIDECAR_C: ArtifactHash =
+    ArtifactHash([6; 32]);
+/// Hash of fake artifact for fake psc-b SP
+pub(super) const ARTIFACT_HASH_SP_PSC_B: ArtifactHash = ArtifactHash([9; 32]);
+/// Hash of fake artifact for fake psc-c SP
+pub(super) const ARTIFACT_HASH_SP_PSC_C: ArtifactHash = ArtifactHash([10; 32]);
+/// Hash of fake artifact for fake gimlet RoT slot A
+pub(super) const ARTIFACT_HASH_ROT_GIMLET_A: ArtifactHash =
+    ArtifactHash([13; 32]);
+/// Hash of fake artifact for fake gimlet RoT slot B
+pub(super) const ARTIFACT_HASH_ROT_GIMLET_B: ArtifactHash =
+    ArtifactHash([14; 32]);
+/// Hash of fake artifact for fake psc RoT slot A
+pub(super) const ARTIFACT_HASH_ROT_PSC_A: ArtifactHash = ArtifactHash([17; 32]);
+/// Hash of fake artifact for fake psc RoT slot B
+pub(super) const ARTIFACT_HASH_ROT_PSC_B: ArtifactHash = ArtifactHash([18; 32]);
+/// Hash of fake artifact for fake switch RoT slot A
+pub(super) const ARTIFACT_HASH_ROT_SWITCH_A: ArtifactHash =
+    ArtifactHash([21; 32]);
+/// Hash of fake artifact for fake switch RoT slot B
+pub(super) const ARTIFACT_HASH_ROT_SWITCH_B: ArtifactHash =
+    ArtifactHash([22; 32]);
+
+// unused artifact hashes contained in our fake TUF repo
+const ARTIFACT_HASH_CONTROL_PLANE: ArtifactHash = ArtifactHash([33; 32]);
+const ARTIFACT_HASH_NEXUS: ArtifactHash = ArtifactHash([34; 32]);
+const ARTIFACT_HASH_HOST_OS: ArtifactHash = ArtifactHash([35; 32]);
+
+/// Hash of fake RoT signing keys
+const ROT_SIGN_GIMLET: &str =
+    "1111111111111111111111111111111111111111111111111111111111111111";
+const ROT_SIGN_PSC: &str =
+    "2222222222222222222222222222222222222222222222222222222222222222";
+const ROT_SIGN_SWITCH: &str =
+    "3333333333333333333333333333333333333333333333333333333333333333";
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub(super) enum MgsUpdateComponent {
+    Sp,
+    Rot,
+    RotBootloader,
+    HostOs,
+}
+
+impl From<&'_ PendingMgsUpdateDetails> for MgsUpdateComponent {
+    fn from(value: &'_ PendingMgsUpdateDetails) -> Self {
+        match value {
+            PendingMgsUpdateDetails::Rot { .. } => Self::Rot,
+            PendingMgsUpdateDetails::RotBootloader { .. } => {
+                Self::RotBootloader
+            }
+            PendingMgsUpdateDetails::Sp { .. } => Self::Sp,
+            PendingMgsUpdateDetails::HostPhase1(_) => Self::HostOs,
+        }
+    }
+}
+
+/// Description of a single fake board (sled, switch, or PSC).
+#[derive(Debug)]
+pub(super) struct TestBoard {
+    pub(super) id: SpIdentifier,
+    pub(super) serial: &'static str,
+    pub(super) sp_board: &'static str,
+    pub(super) rot_board: &'static str,
+    pub(super) rot_sign: &'static str,
+}
+
+impl IdOrdItem for TestBoard {
+    type Key<'a> = SpIdentifier;
+
+    fn key(&self) -> Self::Key<'_> {
+        self.id
+    }
+
+    iddqd::id_upcast!();
```

Comment:
Hm, I haven't used this crate before. I should take a good look at it! These tests are cleaning up nicely

Reply:
Yeah, this crate is great. It's @sunshowers's extension of the `idmap` crate I threw together for maps where the key is derived from the value (but critically, `iddqd` allows keys that borrow from the values; `idmap` didn't).